Error while checking build/AliceO2/O2/o2/macOS-arm for 2802149f83d4adf36d7ac28ce55fcf94e1614b86 at 2022-09-12 16:51:
```
## sw/BUILD/O2-latest/log
/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9798/0/CCDB/test/testBasicCCDBManager.cxx:44:19: error: use of undeclared identifier 'HOST_NAME_MAX'
/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9798/0/CCDB/test/testBasicCCDBManager.cxx:45:31: error: use of undeclared identifier 'HOST_NAME_MAX'
/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9798/0/CCDB/test/testCcdbApi.cxx:63:19: error: use of undeclared identifier 'HOST_NAME_MAX'
/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9798/0/CCDB/test/testCcdbApi.cxx:64:31: error: use of undeclared identifier 'HOST_NAME_MAX'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9798/2802149f83d4adf36d7ac28ce55fcf94e1614b86/build_AliceO2_O2_o2_macOS-arm/pretty.html).

--- a/CCDB/src/CcdbApi.cxx
+++ b/CCDB/src/CcdbApi.cxx
@@ -126,7 +126,7 @@ void CcdbApi::init(std::string const& host)
 }
 
 /**
- * Keep only the alphanumeric characters plus '_' plus '/' from the string passed in argument.
+ * Keep only the alphanumeric characters plus '_' plus '/' plus '.' from the string passed in argument.
  * @param objectName
  * @return a new string following the rule enounced above.
  */
@@ -134,7 +134,7 @@ std::string sanitizeObjectName(const std::string& objectName)
 {
   string tmpObjectName = objectName;
   tmpObjectName.erase(std::remove_if(tmpObjectName.begin(), tmpObjectName.end(),
-                                     [](auto const& c) -> bool { return (!std::isalnum(c) && c != '_' && c != '/'); }),
+                                     [](auto const& c) -> bool { return (!std::isalnum(c) && c != '_' && c != '/' && c != '.'); }),
                       tmpObjectName.end());
   return tmpObjectName;
 }

--- a/CCDB/test/testBasicCCDBManager.cxx
+++ b/CCDB/test/testBasicCCDBManager.cxx
@@ -26,18 +26,49 @@
 
 using namespace o2::ccdb;
 
+static string basePath;
+std::string ccdbUrl = "http://ccdb-test.cern.ch:8080";
+bool hostReachable = false;
+
+/**
+ * Global fixture, ie general setup and teardown
+ */
+struct Fixture {
+  Fixture()
+  {
+    CcdbApi api;
+    api.init(ccdbUrl);
+    std::cout << "ccdb url: " << ccdbUrl << std::endl;
+    hostReachable = api.isHostReachable();
+    std::cout << "Is host reachable ? --> " << hostReachable << std::endl;
+    char hostname[_POSIX_HOST_NAME_MAX];
+    gethostname(hostname, _POSIX_HOST_NAME_MAX);
+    basePath = string("Test/") + hostname + "/pid" + getpid() + "/BasicCCDBManager/";
+    std::cout << "Path we will use in this test suite : " + basePath << std::endl;
+  }
+  ~Fixture()
+  {
+    if (hostReachable) {
+      CcdbApi api;
+      api.init(ccdbUrl);
+      api.truncate(basePath + "*");
+      std::cout << "Test data truncated (" << basePath << ")" << std::endl;
+    }
+  }
+};
+BOOST_GLOBAL_FIXTURE(Fixture);
+
 BOOST_AUTO_TEST_CASE(TestBasicCCDBManager)
 {
   CcdbApi api;
-  const std::string uri = "http://ccdb-test.cern.ch:8080";
-  api.init(uri);
+  api.init(ccdbUrl);
   if (!api.isHostReachable()) {
-    LOG(warning) << "Host " << uri << " is not reacheable, abandoning the test";
+    LOG(warning) << "Host " << ccdbUrl << " is not reacheable, abandoning the test";
     return;
   }
   //
-  std::string pathA = "Test/CachingA";
-  std::string pathB = "Test/CachingB";
+  std::string pathA = basePath + "CachingA";
+  std::string pathB = basePath + "CachingB";
   std::string ccdbObjO = "testObjectO";
   std::string ccdbObjN = "testObjectN";
   std::map<std::string, std::string> md;
@@ -48,7 +79,7 @@ BOOST_AUTO_TEST_CASE(TestBasicCCDBManager)
 
   // test reading
   auto& cdb = o2::ccdb::BasicCCDBManager::instance();
-  cdb.setURL(uri);
+  cdb.setURL(ccdbUrl);
   cdb.setTimestamp((start + stop) / 2);
   cdb.setCaching(true);
 

--- a/CCDB/test/testCcdbApi.cxx
+++ b/CCDB/test/testCcdbApi.cxx
@@ -60,7 +60,9 @@ struct Fixture {
     cout << "ccdb url: " << ccdbUrl << endl;
     hostReachable = api.isHostReachable();
     cout << "Is host reachable ? --> " << hostReachable << endl;
-    basePath = string("Test/pid") + getpid() + "/";
+    char hostname[_POSIX_HOST_NAME_MAX];
+    gethostname(hostname, _POSIX_HOST_NAME_MAX);
+    basePath = string("Test/TestCcdbApi/") + hostname + "/pid" + getpid() + "/";
     cout << "Path we will use in this test suite : " + basePath << endl;
   }
   ~Fixture()
@@ -377,8 +379,6 @@ BOOST_AUTO_TEST_CASE(list_test, *utf::precondition(if_reachable()))
 
   // more complex tree
   TH1F h1("object1", "object1", 100, 0, 99);
-  cout << "storing object 1 in Test" << endl;
-  f.api.storeAsTFile(&h1, "Test", f.metadata);
   cout << "storing object 2 in Test/Detector" << endl;
   f.api.storeAsTFile(&h1, basePath + "Detector", f.metadata);
   cout << "storing object 3 in Test/Detector" << endl;

--- a/CCDB/test/testCcdbApi_ConfigParam.cxx
+++ b/CCDB/test/testCcdbApi_ConfigParam.cxx
@@ -61,8 +61,9 @@ struct Fixture {
     api.init(ccdbUrl);
     cout << "ccdb url: " << ccdbUrl << endl;
     hostReachable = api.isHostReachable();
-    cout << "Is host reachable ? --> " << hostReachable << endl;
-    basePath = string("Test/pid") + getpid() + "/";
+    char hostname[_POSIX_HOST_NAME_MAX];
+    gethostname(hostname, _POSIX_HOST_NAME_MAX);
+    basePath = string("Test/") + hostname + "/pid" + getpid() + "/";
     cout << "Path we will use in this test suite : " + basePath << endl;
   }
   ~Fixture()

